.github/workflows/CI.yaml

```diff
@@ -0,0 +1,30 @@
+name: CI
+
+on: [push]
```

Comment:
Should this be run on pull requests against `main` instead of any push?

Reply:
I think CI should run at every commit for build, checks and test

in theory we could run on main stuff like deploy or more heavy checks